### DISCUSSION TITLE
Should we drop support for stable-1 and/or stable-2

### DISCUSSION THREAD

**Discussion by laurent-indermuehle** [ansible-collections/ansible.mysql#537] [Polls]
Backporting every bug fix can be very time-consuming, especially when merge conflicts arise.

If you're a user of the 1.x or 2.x branch, please participate in the poll to let us know if you want us to continue supporting them.

Otherwise, I suggest that we only focus on the 3.x branch to ease the burden on the maintainers.

**Comment by Andersson007** [comment]
Please folks vote @betanummeric  @rsicart @Jorge-Rodriguez @bmalynovytch 

@laurent-indermuehle thanks for opening the pole.

I agree that maintaining both can feel like a burden.
I'm personally +1 to dropping `stable-1` and keeping `stable-2` (not to push users too hard to use the latest version still providing bugfixes).

**Comment by betanummeric** [comment]
From a user perspective, I would like a policy on how long (in time) major releases are supported.
1.0.0 was released on 2020-08-17 (~2y9m ago)
2.0.0 was released on 2021-04-15 (~2y1m ago)
3.0.0 was released on 2021-12-01 (~1y5m ago)
We could promise to support every major release for 2/2.5/3 years.
How about that (numbers are up for discussion)?

**Reply by Andersson007** [reply]
We can see support for @betanummeric 's idea.
Should we open another vote about the support time?
Keeping things simple, if you folks OK with **2 years**, please thumb this mine comment up OR open another vote with 1, 2, 3, .. years as options.

**Reply by laurent-indermuehle** [reply]
I'm wondering what will happen if we release 4.x.y after 3.x.y is EOL. We will terminate that branch immediately?

**Reply by Andersson007** [reply]
@laurent-indermuehle good question. I think it makes sense to maintain it because some new features can be introduced right before a new major release. Should we then maintain the previous stable version (backport bugfixes) for 2 years since the next major release? I updated my code change suggestion https://github.com/ansible-collections/community.mysql/pull/543#discussion_r1184956080 correspondingly.

**Reply by markuman** [reply]
I think a major release must only be released when there are breaking changes. Otherwise there is no need for a major release.  
Basically I like the policy that  (_of course_) the current and the previous major release are supported.  
So when 4.X will be released, 2.X becomes EOL and 3.X will receive only bug- and security fixes (_mostly_).

**Comment by markuman** [comment]
+1 for keep both.  
there is no need to drop them, even if it's EOL and is unused by us.  
but that keeps the opportunity that a user, that rely on 1.X release (_for whatever reason_), can still checkout stable-1 branch and do what every they need to do.  

Other big collections keeps the branches also, like community.general or community.aws

**Reply by betanummeric** [reply]
From my understanding, nobody intends to delete any release branch. The discussion is about where patches (new commits) are needed/promised.

**Reply by Andersson007** [reply]
Yeah, the latest suggestion is to keep support for 2 years since the next major release date, i.e. the current version is 3 and when we release 4, we'll be keeping backporting fixes to stable-3 for two years after that. Then will stop.

**Reply by laurent-indermuehle** [reply]
I should have written "Drop support" and "Keep maintaining" for more clarity. Still learning how to be explicit, sorry :(
